scaleway/resource_instance_private_nic.go

```diff
@@ -81,7 +81,7 @@ func resourceScalewayInstancePrivateNICRead(ctx context.Context, d *schema.Resou
 		return diag.FromErr(err)
 	}
 
-	res, err := instanceAPI.GetPrivateNIC(&instance.GetPrivateNICRequest{
+	res, err := instanceAPI.WaitForPrivateNIC(&instance.WaitForPrivateNICRequest{
```

Comment:
Should be replaced by a waitForPrivateNIC using the same pattern that was recently merged that allow for quick retries